src/Packagist/WebBundle/Controller/ApiController.php

```diff
@@ -57,6 +57,39 @@ public function packagesAction(Request $req)
     }
 
     /**
+     * @Route("/api/create-package", name="generic_create", defaults={"_format" = "json"})
+     * @Method({"POST"})
+     */
+    public function createPackageAction(Request $request)
+    {
+
+        $payload = json_decode($request->getContent(), true);
+        if (!$payload) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Missing payload parameter'), 406);
+        }
+        $url = $payload['repository']['url'];
+        $package = new Package;
+        $package->setEntityRepository($this->getDoctrine()->getRepository('PackagistWebBundle:Package'));
+        $package->setRouter($this->get('router'));
+        $user = $this->findUser($request);
+        $package->addMaintainer($user);
+        $package->repository = $url;
+        if ($this->get('validator')->validate($package)) {
+            return new JsonResponse(array('status' => 'error', 'message' => 'Invalid package'), 406); 
```

Comment:
no, because the ConstraintViolationList does not implement JsonSerializable, and so it would be json-encoded as an empty JS object (there is no public properties)